File: 0234-palindrome-linked-list/0234-palindrome-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    bool isPalindrome(ListNode* head) {
        ListNode* temp = head; 
        stack<int>st; 
        while( temp!= NULL){
            st.push(temp->val); 
            temp = temp->next; 
        }
        temp = head; 
        while( temp!= NULL){
            if( st.top() != temp->val) return false; 
            temp = temp->next; 
            st.pop(); 
        }
        return true;
    }
};

File: 0234-palindrome-linked-list/0234-palindrome-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public boolean isPalindrome(ListNode head) {
        ListNode temp = head; 
        Stack<Integer>st = new Stack <>();

        while( temp!= null ){
            st.push(temp.val); 
            temp = temp.next; 
        } 
        temp = head; 
        while( temp!= null){
            if( st.peek() !=  temp.val) return false ;  
            temp =temp.next; 
            st.pop(); 
        }       
        return true;
    }
}